### PR TITLE
Fixes #10 - Get each day individually instead of entire set

### LINKED ISSUE #10: Retrieve only one day at a time, adding to previous date
For the zoom api, it looks like you can only be retrieving a single day at a time, so set the start date in the config and only go from that.

If a previous csv file exists, just use that as the date to start from instead of the configured date.

### PR DESCRIPTION
**Pull request by jonespm**
Retrieves every day individually in a loop
Switch config from json to yaml to add comments to the configuration
Optimization to not create append to a dataframe until the end

### CODE REVIEW

File: report.py [hunk 1]
```diff
@@ -56,34 +35,79 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}):
...
+    params['page_size'] = 300
+    total_list = []
+    # TODO: Detect the date from the previous CSV
+    # Either loop for all dates or just one a single report
+    if use_date:
+        early_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-03-01')).date()
+        # Only use the date as a paramter
+        # Loop until yesterday (this will go until now() -1)
+        for i in range((datetime.now().date() - early_date).days):
+            param_date = early_date + timedelta(days=i)
```
Comment (zqian) [inline report.py:76]: The "from" and "to" params are in local time, and the return results are in GMT format, right?

File: report.py [hunk 2]
```diff
@@ -56,34 +35,79 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}):
...
+        total_list.extend(zoom_loop(url, headers, json_attribute_name, dict(params), page_token))
+    # output csv file
+    total_df = pd.DataFrame(total_list)
+    total_df.index.name = "index_id"
+    output_file_name = f"total_{json_attribute_name}.csv"
+    # Remove any duplicate uuids in the record
+    logger.info(f"Initial dataframe size: {len(total_df)}")
+    if "uuid" in total_df:
+        total_df.drop_duplicates("uuid", inplace=True)
+        logger.info(f"Dataframe with duplicates removed: {len(total_df)}")
```
Comment (zqian) [inline report.py:91]: Would be interesting to see the number difference with and without the duplicates.

File: report.py [hunk 3]
```diff
@@ -56,34 +35,79 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}):
...
-    f.args = params
-    print(f.url)
 
-    # get total page count
-    total_page_count = get_total_page_count(f"{url}{f.url}", headers)
+def run_report(api_url: str, headers: dict, json_attribute_name: str,
+               default_params: dict = {}, page_token: bool = False, use_date: bool = False):
+    url = ZOOM_BASE_URL+api_url
+    params = default_params
+    params['page_size'] = 300
```
Comment (zqian) [inline report.py]: can use fixed page_size for all calls?

Comment (jonespm) [reply report.py]: Yeah, all of these were 300, I can switch this to a parameter though just incase.

File: report.py (hunk 1 above)
Comment (jonespm) [reply report.py:76]: Yeah, I'm only iterating through "dates" so like 2020-03-25. The API currently only accepts this format and doesn't allow for times. The results are in GMT so it's looks like it's GMT all around. The data returned is currently slightly delayed and only includes data up until yesterday currently. 

We could probably have a more frequent process that just runs on the current day without inserting duplicates. I could create this as a new future issue.

File: report.py (hunk 2 above)
Comment (jonespm) [reply report.py:91]: Yeah, I'm seeing usually only seeing a few (3-8 duplicates). We could probably display the duplicates and I'm not sure why they're created, maybe something for the future.

**Comment by jonespm** [comment]
Going to merge this so I can get it migrated over to the other project

**Pull request merged**